src/app/vm/vm-creation/services/vm-deployment.service.ts

```diff
@@ -162,6 +164,13 @@ export class VmDeploymentService {
       .switchMap(() => {
         return this.userTagService.getSavePasswordForAllVms();
       })
+      .switchMap(() => {
+        if (state.template.agreementAccepted) {
+          return this.vmTagService.setAgreement(vm)
+        } else {
+          return Observable.of(null);
+        }
+      })
       .switchMap((tag) => {
```

Comment:
looks like you have a bug here.
If agreement has been accepted then a VM password will be saved